advice/views.py

```diff
@@ -45,60 +49,68 @@ def put(self, request, id):
+        return Response(data=data, status=status.HTTP_401_UNAUTHORIZED)
 
-        if request.method == "DELETE":
-            result = advice.delete()
+    try:
+        advice = AdviceModel.objects.filter(advice_id__in=del_id["id"])
+    except AdviceModel.DoesNotExist:
+        data["success"] = False
+        data["msg"] = "Record does not exist"
+        data["data"] = []
+        return Response(data=data, status=status.HTTP_401_UNAUTHORIZED)
 
-            AdviceGroupAdviceModel.objects.filter(advicemodel_id__in=del_id["id"]).delete()
-            data["success"] = True
-            data["msg"] = "Data deleted successfully."
-            data["deleted"] = result
-            return Response(data=data, status=status.HTTP_200_OK)
+    if request.method == "DELETE":
+        result = advice.delete()
+
+        AdviceGroupAdviceModel.objects.filter(advicemodel_id__in=del_id["id"]).delete()
+        data["success"] = True
+        data["msg"] = "Data deleted successfully."
+        data["deleted"] = result
+        return Response(data=data, status=status.HTTP_200_OK)
 
     # ================= Create New Record=========================
-    def post(self, request):
-        data = {}
-        if request.method == "POST":
-            advice = AdviceModel()
-            serializer = AdviceSerializers(advice, data=request.data)
 
 
-            if serializer.is_valid():
-                serializer.save()
-                if "advice_group_name" in request.data:
-                    insert_advice_group(request, serializer.data["advice_id"])
+@api_view(["POST"])
+@authentication_classes([JWTAuthentication])
+@validate_permission("advice", "add")
+def create(request):
+    data = {}
+    if request.method == "POST":
+        advice = AdviceModel()
+        serializer = AdviceSerializers(advice, data=request.data)
 
-                data["success"] = True
-                data["msg"] = "Data updated successfully"
-                data["data"] = serializer.data
-                return Response(data=data, status=status.HTTP_201_CREATED)
+        if serializer.is_valid():
+            serializer.save()
+            if "advice_group_name" in request.data:
+                insert_advice_group(request, serializer.data["advice_id"])
 
-            data["success"] = False
-            data["msg"] = serializer.errors
+            data["success"] = True
+            data["msg"] = "Data updated successfully"
             data["data"] = serializer.data
-            return Response(data=data, status=status.HTTP_400_BAD_REQUEST)
+            return Response(data=data, status=status.HTTP_201_CREATED)
+
+        data["success"] = False
+        data["msg"] = serializer.errors
+        data["data"] = serializer.data
+        return Response(data=data, status=status.HTTP_400_BAD_REQUEST)
 
 
-@api_view(['POST'])
+@api_view(["POST"])
 @authentication_classes([JWTAuthentication])
-@permission_classes([IsAuthenticated])
+@validate_permission("advice", "change")
```

Comment:
where function has two parameter @validate_permission_id should be used. import the function from the same file.